lib/gem-miner/github_client.rb

```
@@ -0,0 +1,38 @@
+require 'base64'
+require 'octokit'
+
+module GemMiner
+  # A Github client with GemMiner-specific extensions.
+  class GithubClient
+
+    def initialize(github_access_token = nil)
+      @internal_client = Octokit::Client.new(
+        auto_paginate: true,
+        access_token: github_access_token
+      )
```

I would raise an error if no github_access_token is passed